improver_tests/utilities/test_spatial.py

```diff
@@ -253,6 +262,23 @@ def test_lat_lon_not_equal_spacing(self):
         with self.assertRaisesRegex(ValueError, msg):
             calculate_grid_spacing(self.lat_lon_cube, "degrees", rtol=self.rtol)
 
+    def test_lat_lon_equal_spacing_recurring_decimal_spacing_fails(self):
+        """Test grid spacing with lat-lon grid with with 1/3 degree
+        intervals with tolerance of 1.0e-5"""
+        self.lat_lon_cube.coord("longitude").points = self.longitude_points_thirds
+        msg = "Coordinate longitude points are not equally spaced"
+        with self.assertRaisesRegex(ValueError, msg):
+            calculate_grid_spacing(self.lat_lon_cube, "degrees", rtol=self.rtol)
+
+    def test_lat_lon_equal_spacing_recurring_decimal_spacing_passes(self):
+        """Test grid spacing outputs with lat-lon grid with 1/3 degree
+        intervals with tolerance of 3.0e-5"""
+        self.lat_lon_cube.coord("longitude").points = self.longitude_points_thirds
```

Comment:
This test made me a bit suspicious because it occurred to me that we are replacing a numpy array with a list (most of the other tests in this class do the same thing). If I change the definition to be a numpy array (dtype=float32), then this test fails - unless I increase `self.rtol_thirds` to 4e-5 or allow the precision of the array to be float64.

As we want this test to replicate real data (your Australian data), then I think we want the longitude points to be float32, which means that the tolerance needs to be increased, and hence the tolerance used in grid.py needs to be increased too. Please can you modify the test to do this - I'd suggest changing BOTH longitude lists to be float32 numpy arrays, although this requires changing the call to `reverse()` in `test_lat_lon_negative_spacing` as well, AND increasing the tolerance values as suggested.

Reply:
@MoseleyS  
Thanks for checking.  I made changes accordingly.